.github/workflows/ci.yml

```diff
@@ -383,6 +383,10 @@ jobs:
           package-generators: ${{ matrix.mrdocs-package-generators }}
           package-artifact: false
 
+      # If this fails, the config was updated without updating the schema
+      - name: Check YAML schema
+        run: git --no-pager diff --exit-code -- docs/mrdocs.schema.json
```

Comment:
I'm sorry for my ignorance. How will this command know if the config JSON file changed? 

As far as I understand, this command checks for changes only in the `docs/mrdocs.schema.json` file and compares it to the last committed version. But this file not changing is not an error by itself.

Also, if I understand correctly, `docs/mrdocs.schema.json` only contains a subset of the config file so not all changes in the config file require changes in `docs/mrdocs.schema.json`. For instance, if some field changes from `string` to `path`, `docs/mrdocs.schema.json` will not change.

Reply:
The idea is that `generate-config-info.py` runs during the build and updates `docs/mrdocs.schema.json`. Then, after the build, CI checks if the file was updated (i.e. the schema changed, but wasn't committed).

Reply:
Oh. OK. So, it should return 0 even if the contents are the same.

In CI, we're also having an issue with checkout not constantly downloading the source files from the repository. This is fixable by setting up git first. 

Or we could remove this test since it does not test much—only that the file has been touched.

Reply:
> In CI, we're also having an issue with checkout not constantly downloading the source files from the repository. This is fixable by setting up git first.
> 
> Or we could remove this test since it does not test much—only that the file has been touched.

Then it's easier to have a separate script, and remove the test from CI, I think.

Reply:
Mmm... Yes. We don't strictly need a separate script, but that's the best idea (Unless the logic for the schema reuses a lot of the logic for generating the C++ files but I doubt that.) because the CMake scripts would be generating a file we don't need to compile the library. 

I'm looking at https://www.schemastore.org/json/, and they suggest a [MegaLinter](https://megalinter.github.io/) action to check files. It could replace the whole step. The only problem is this MegaLinter is way more than we need.

Reply:
> I'm looking at [schemastore.org/json](https://www.schemastore.org/json/), and they suggest a [MegaLinter](https://megalinter.github.io/) action to check files. It could replace the whole step. The only problem is this MegaLinter is way more than we need.

[ajv](https://ajv.js.org/guide/environments.html#command-line-interface) can validate YAML files. We use it [at Chatterino to validate JSON files in CI](https://github.com/Chatterino/chatterino2/blob/c0a5a3e8058255376494408730c9948aae8f66af/.github/workflows/lint.yml#L30-L33). Here, the command-line would be

```sh
npx ajv validate -s mrdocs.schema.json -d mrdocs.yml
```

Through this, it also validates the schema. Seems it doesn't like `https://` in the schema, but expects `http://`.

Reply:
Yes. This could make a better GHA step here. Using npx makes it all much more manageable.

However, we want to validate mrdocs.schema.json against the meta schema https://json-schema.org/draft/2020-12/schema.